routing/missioncontrol.go

```diff
@@ -373,7 +373,7 @@ func (m *MissionControl) GetHistorySnapshot() *MissionControlSnapshot {
 // reportPaymentFail reports a failed payment to mission control as input for
 // future probability estimates. It returns a bool indicating whether this error
 // is a final error and no further payment attempts need to be made.
-func (m *MissionControl) reportPaymentFail(rt *route.Route,
+func (m *MissionControl) reportPaymentFail(paymentID uint64, rt *route.Route,
```

Comment:
Why does it need the payment ID? Atm this is a lower level identifier that shouldn't really impact the operation of mission control.

Reply:
It is to prevent the same payment result from being recorded multiple times because of `lnd` restarts. It makes the call idempotent.